pkg/controller/storagecluster/reconcile.go

```diff
@@ -590,6 +597,21 @@ func (r *ReconcileStorageCluster) ensureCephConfig(sc *ocsv1.StorageCluster, req
 // ensureCephCluster ensures that a CephCluster resource exists with its Spec in
 // the desired state.
 func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// if StorageClass is "gp2" or "io1" based, set tuneSlowDeviceClass to true
+	// this is for performance optimization of slow device class
+	//TODO: If for a StorageDeviceSet there is a separate metadata pvc template, check for StorageClass of data pvc template only
+	for i, ds := range sc.Spec.StorageDeviceSets {
+		throttle, err := r.throttleStorageDevices(*ds.DataPVCTemplate.Spec.StorageClassName)
```

Comment:
I think for correctness, this should be done on a per-OSD-PV basis, not for the DataPVCTemplate: In theory (even though it would not be recommended), we could add completely different types of disks when we expand the storage, right? (This is what I have also been criticizing - I think we should offer a choice of the storage class upon initial creation, and then lock the storage class for future expansion of the created pool...)

Reply:
I not sure I understand the whole `StorageClassDeviceSet` design completely.
But, based on current implementation, `DataPVCTemplate` can have only one `PersistentVolumeClaimTemplate` which means a Device Set can have only similar devices (based on mentioned `Storage Class`). And this information is enough to decide on throttling (which is eventually handled by rook implementation).

We could have a mix of devices (currently possible only by having different DeviceSets) and separation of Metadata and Data PVCs which is why leaving a TODO comment to pick up once we have all required things based on new implementation.

@jarrpa and I had a long conversation on this and he approved of these changes.